Framework/PythonInterface/plugins/algorithms/IntegratePeaksShoeboxTOF.py

```diff
@@ -366,6 +368,13 @@ def PyExec(self):
                 ipos_predicted = [peak_data.irow, peak_data.icol, ix]
                 det_edges = peak_data.det_edges if not integrate_on_edge else None
 
+                # get all detector IDs related to the spectrum of the peak
+                ispec = ispecs[peak_data.irow, peak_data.icol]
+                detectors_per_spec = ws.getSpectrum(int(ispec)).getDetectorIDs()
```

Comment:
Sorry, this will get all the detector IDs associated with the spectrum at the predicted peak position.
Here we want to save all the detector IDs that contribute to the peak, which we only know after the integration (only need one per spectrum I think - so just take first one in case of WISH).
In this case this is something like this (note on the PeakData class saves the detector IDs I think, so no need to retrieve them from the spectrum)
```
peak_detids = peak_data.detids[ipos[0]-nrows//2:ipos[0]+nrows//2, ipos[1]-ncols//2:ipos[1]+ncols//2, ipos[2]-nbins//2:ipos[2]+nbins//2]
```
Where `ipos` is a vector of the optimised shoebox center indices and `nrows`, `ncols`,`nbins` are the dimensions of the peak region (the positive bit of the convolution kernel).
You might have to get these from the `ShoeboxResult` object as the two-step integration makes the exec a bit messy (in which case you might have to make these result objects even if no output file is requested). Note if a peak hasn't been integrated it shouldn't have a peak shape set (like the behaviour of `IntegratePeaksMD` etc.)

Reply:
The TOF extent is same for all detectors in this algorithm (and in skew) - it is given by `nbins` like so
`ws.readX(ispec)[ipos[2] + nbins//2] - ws.readX(ispec)[ipos[2] - nbins//2]`